middlewares/auth/authenticator.go

```diff
@@ -31,44 +32,54 @@ func NewAuthenticator(authConfig *types.Auth) (*Authenticator, error) {
 			return nil, err
 		}
 		basicAuth := goauth.NewBasicAuthenticator("traefik", authenticator.secretBasic)
-		authenticator.handler = negroni.HandlerFunc(func(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
-			if username := basicAuth.CheckAuth(r); username == "" {
-				log.Debug("Basic auth failed...")
-				basicAuth.RequireAuth(w, r)
-			} else {
-				log.Debug("Basic auth success...")
-				if authConfig.HeaderField != "" {
-					r.Header[authConfig.HeaderField] = []string{username}
-				}
-				next.ServeHTTP(w, r)
-			}
-		})
+		authenticator.handler = tracingMiddleware.NewNegroniHandlerWrapper("Auth Basic", createAuthBasicHandler(basicAuth, authConfig), false)
```

Comment:
tracingMiddleware could be nil ?

Reply:
`tracingMiddleware can not be nil because there is a default value https://github.com/containous/traefik/pull/2587/files/f7b0da15af1a7d836f9e21cae6f252b4580c3c7e#diff-f80800db6a453c4a7fff51b05de444f2R209`